openlibrary/catalog/get_ia.py

```diff
@@ -298,19 +247,3 @@ def marc_formats(ia, host=None, path=None):
         if all(has.values()):
             break
     return has
-
-def test_get_ia():
```

Comment:
Oops, removed this test too early. `get_marc_record_from_ia()` should be able to get `poeticalworksoft00grayiala`, handle the bad XML and then load the binary MARC, and return that.

Reply:
This is not the current behaviour. Because `MarcXml` provides a generator for the fields it doesn't validate or know there are some empty fields in the data at init, so won't fail at this point. `poeticalworksoft00grayiala` can be loaded as XML, and will raise a `BlankTag` exception when the field is encountered.

Current behaviour is to only skip to Binary MARC if the XML file does not exist, or there is an problem loading the XML file. It does not validate the MARCXML format (just basic XML).